jgiven-core/src/main/java/com/tngtech/jgiven/impl/inject/ValueInjector.java

```diff
@@ -90,6 +90,8 @@ public void updateValues( Object object ) {
                 if( value != null ) {
                     field.set( object, value );
                     log.debug( "Setting field {} to value {}", field, value );
+                } else if( isRequired( field ) ) {
+                    throw new RuntimeException( "TODO Better exception?" );
```

Comment:
Yeah, this should be a different exception. The question is which kind of exception. `JGivenWrongUsageException` is not really fitting, I think, because JGiven is not really used in a wrong way in this case. JGivenPreconditionException/JGivenAssumptionException/JGivenUserException maybe?